deeplearning4j/deeplearning4j-nlp-parent/deeplearning4j-nlp/src/main/java/org/deeplearning4j/models/embeddings/learning/impl/elements/CBOW.java

```diff
@@ -178,133 +178,223 @@ public boolean isEarlyTerminationHit() {
+                        .codes(codes)
+                        .indices(points)
+                        .preciseMode(configuration.isPreciseMode())
+                        .alpha(alpha)
+                        .inferenceVector( inferenceVector != null ? inferenceVector : Nd4j.empty(syn0.get().dataType()))
+                        .numWorkers(workers)
+                        .iterations(useInference ? configuration.getIterations() * configuration.getEpochs() : 1)
+                        .build();
+
+            }
+            else if (useHS) {
+                cbowInference = CbowInference.builder()
+                        .target(currentWord.getIndex())
+                        .ngStarter(currentWord.getIndex())
+                        .randomValue((int) randomValue)
+                        .context(windowWords)
+                        .expTable(expTable.get())
+                        .lockedWords(inputStatuses)
+                        .syn0(syn0.get())
+                        .syn1(syn1.get())
+                        .syn1Neg(Nd4j.empty(syn0.get().dataType()))
+                        .negTable(table.get() == null ? Nd4j.empty(syn0.get().dataType()) : table.get())
+                        .codes(codes)
+                        .indices(new int[0])
+                        .numLabels(numLabels)
+                        .alpha(alpha)
+                        .preciseMode(configuration.isPreciseMode())
+                        .inferenceVector(inferenceVector != null ? inferenceVector : Nd4j.empty(syn0.get().dataType()))
+                        .numWorkers(workers)
+                        .iterations(useInference ? configuration.getIterations() * configuration.getEpochs() : 1)
+                        .build();
+
+            }
+            else if (useNegative) {
+                cbowInference = CbowInference.builder()
+                        .target(currentWord.getIndex())
+                        .ngStarter(currentWord.getIndex())
+                        .randomValue((int) randomValue)
+                        .context(windowWords)
+                        .expTable(expTable.get())
+                        .negTable(table.get())
+                        .lockedWords(inputStatuses)
+                        .syn0(syn0.get())
+                        .syn1(Nd4j.empty(syn0.get().dataType()))
+                        .syn1Neg(Nd4j.empty(syn0.get().dataType()))
+                        .codes(new byte[0])
+                        .indices(new int[0])
+                        .numLabels(numLabels)
+                        .alpha(alpha)
+                        .preciseMode(configuration.isPreciseMode())
+                        .inferenceVector(inferenceVector != null ? inferenceVector : Nd4j.empty(syn0.get().dataType()))
+                        .numWorkers(workers)
+                        .iterations(useInference ? configuration.getIterations() * configuration.getEpochs() : 1)
+                        .build();
+            }
```

Comment:
All of those builder repetitions look fishy. Why not create the builder with the common options set and then set the options that are actually different?